packages/react/src/vertex4.tsx

```diff
@@ -0,0 +1,322 @@
+    categoryID?: string;
+    text: string;
+    textHeight?: number;
+    textPadding?: number;
+    textboxStrokeWidth?: number;
+    icon?: Icon;
+    annotations?: Icon[];
+    iconAnnotations?: Icon[];
+    annotationsHeight?: number;
+    annotationGutter?: number;
+    textFill?: string;
+    textboxFill?: string;
+    textboxStroke?: string;
+    textFontFamily?: string;
+    cornerRadius?: number;
+    subText?: any;
+    onSizeUpdate?: (size: { width: number, height: number }) => void;
+    showLabel?: boolean;
+    noLabelRadius?: number;
+    iconBorderWidth?: number;
+    iconBorderColor?: string;
+    iconBackgroundColor?: string;
+    shapeOffsetX?: number;
+    shapeOffsetY?: number;
+    iconOffsetX?: number;
+    iconOffsetY?: number;
+    iconPadding?: number;
+    iconFontSize?: number;
+    iconFontColor?: string;
+    iconFontFamily?: string;
+    iconText?: string;
+    shapeRendering?: "auto" | "optimizeSpeed" | "crispEdges" | "geometricPrecision";
+}
+
+export const Vertex4: React.FunctionComponent<IVertex4> = ({
+    categoryID = "",
+    text = "",
+    textHeight = 10,
+    textPadding = 4,
+    textFill = "#287EC4",
+    textboxFill = "white",
+    textboxStroke = "#CCCCCC",
+    textboxStrokeWidth = 1,
+    textFontFamily = "Verdana",
+    annotationGutter = 2,
+    annotations = [],
+    iconAnnotations = [],
+    cornerRadius = 3,
+    icon = {},
+    subText = {},
+    showLabel = true,
+    noLabelRadius = 5,
+
+    iconBorderWidth = 1,
+    iconBorderColor = "#333",
+
+    iconBackgroundColor = "#fff",
+
+    iconFontColor = "#000",
+    iconFontSize=20,
+    iconFontFamily="FontAwesome",
+
+    shapeOffsetX = 0,
+    shapeOffsetY = 0,
+    iconOffsetX = 0,
+    iconOffsetY = 0,
+
+    iconPadding = 4,
+    iconText="?",
+    shapeRendering="auto"
+}) => {
+    icon = {
+        height: 50,
+        imageChar: "?",
+        imageFontFamily: "FontAwesome",
+        imageCharFill: "#555555",
+        fill: "transparent",
+        strokeWidth: 0,
+        ...icon
+    };
+    subText = {
+        text: "",
+        fill: "white",
+        textFill: "#555555",
+        ...subText
+    };
+
+    const annoOffsetY = 0;
+    const labelWidth = React.useMemo(() => {
+        return Utility.textSize(text, textFontFamily, textHeight, false).width;
+    }, [text, textFontFamily, textHeight]);
+
+    let labelShapeWidth = 0;
+    if (text !== "") {
+        labelShapeWidth = labelWidth + (textPadding * 2) + (textboxStrokeWidth * 2);
+    }
+    let fullAnnotationWidth = labelShapeWidth + annotationGutter;
+    const textOffsetX = fullAnnotationWidth - (labelShapeWidth / 2);
+    
+    const textShapeHeight = textHeight + (annotationGutter*2) + (textboxStrokeWidth * 2);
+    const annoWidthArr = React.useMemo(() => {
+        return annotations.map((anno, i) => {
+            return Utility.textSize(anno.imageChar, anno.imageFontFamily, anno.height, false).width;
+        });
+    }, [annotations]);
```

Comment:
This will only update if the annotations array reference changes, not if the content of the annotations array changes.

Reply:
@jbrundage bump